From 5d504694380191d85345632caa5afd9d39c4ad99 Mon Sep 17 00:00:00 2001
From: Andrew Jeddeloh <ajeddelo@redhat.com>
Date: Mon, 22 Jul 2019 11:09:58 -0700
Subject: [PATCH 2/2] providers/aws: get region after getting config.

We try to set the region as soon as Ignition starts, but sometimes the
metadata server isn't accessible yet. Set the region after we've fetched
the config since the server must be up by then.
---
 internal/providers/ec2/ec2.go | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/internal/providers/ec2/ec2.go b/internal/providers/ec2/ec2.go
index 535398e..65c998a 100644
--- a/internal/providers/ec2/ec2.go
+++ b/internal/providers/ec2/ec2.go
@@ -48,6 +48,13 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 		return types.Config{}, report.Report{}, err
 	}
 
+	// Determine the partition and region this instance is in
+	regionHint, err := ec2metadata.New(f.AWSSession).Region()
+	if err != nil {
+		regionHint = "us-east-1"
+	}
+	f.S3RegionHint = regionHint
+
 	return util.ParseConfig(f.Logger, data)
 }
 
@@ -58,14 +65,8 @@ func NewFetcher(l *log.Logger) (resource.Fetcher, error) {
 	}
 	sess.Config.Credentials = ec2rolecreds.NewCredentials(sess)
 
-	// Determine the partition and region this ec2 is in
-	regionHint, err := ec2metadata.New(sess).Region()
-	if err != nil {
-		regionHint = "us-east-1"
-	}
 	return resource.Fetcher{
-		Logger:       l,
-		AWSSession:   sess,
-		S3RegionHint: regionHint,
+		Logger:     l,
+		AWSSession: sess,
 	}, nil
 }
-- 
2.21.0

